src/uuid.ts

```diff
@@ -1,57 +1,214 @@
+   */
+  equals(otherId: string | Buffer | UUID): boolean {
+    if (!otherId) {
+      return false;
+    }
+
+    if (otherId instanceof UUID) {
+      return otherId.id.equals(this.id);
+    }
 
-  // Parse ########-....-....-....-............
-  arr[0] = (v = parseInt(uuid.slice(0, 8), 16)) >>> 24;
-  arr[1] = (v >>> 16) & 0xff;
-  arr[2] = (v >>> 8) & 0xff;
-  arr[3] = v & 0xff;
+    try {
+      return new UUID(otherId).id.equals(this.id);
+    } catch {
+      return false;
+    }
+  }
 
-  // Parse ........-####-....-....-............
-  arr[4] = (v = parseInt(uuid.slice(9, 13), 16)) >>> 8;
-  arr[5] = v & 0xff;
+  /**
+   * Checks if a value is a valid bson UUID
+   * @param input - UUID, string or Buffer to validate.
+   */
+  static isValid(input: string | Buffer | UUID): boolean {
+    if (!input) {
+      return false;
+    }
 
-  // Parse ........-....-####-....-............
-  arr[6] = (v = parseInt(uuid.slice(14, 18), 16)) >>> 8;
-  arr[7] = v & 0xff;
+    if (input instanceof UUID) {
+      return true;
+    }
 
-  // Parse ........-....-....-####-............
-  arr[8] = (v = parseInt(uuid.slice(19, 23), 16)) >>> 8;
-  arr[9] = v & 0xff;
+    if (typeof input === 'string') {
+      return uuidHexStringValidateV4(input);
+    }
 
-  // Parse ........-....-....-....-############
-  // (Use "/" to avoid 32-bit truncation when bit-shifting high-order bytes)
-  arr[10] = ((v = parseInt(uuid.slice(24, 36), 16)) / 0x10000000000) & 0xff;
-  arr[11] = (v / 0x100000000) & 0xff;
-  arr[12] = (v >>> 24) & 0xff;
-  arr[13] = (v >>> 16) & 0xff;
-  arr[14] = (v >>> 8) & 0xff;
-  arr[15] = v & 0xff;
+    if (Buffer.isBuffer(input)) {
+      // check for length & uuid version (https://tools.ietf.org/html/rfc4122#section-4.1.3)
+      return input.length === BYTE_LENGTH && (input[6] & 0x40) === 0x40;
+    }
+
+    return false;
+  }
 
-  return arr;
+  /** @internal */
+  toExtendedJSON(options: EJSONOptions = {}): BinaryExtendedLegacy | BinaryExtended {
+    const binary = new Binary(this.id, Binary.SUBTYPE_UUID);
+    return binary.toExtendedJSON(options);
+  }
+
+  /** @internal */
+  static fromExtendedJSON(
+    doc: UUIDExtended | BinaryExtended | BinaryExtendedLegacy,
+    options: EJSONOptions = {}
+  ): UUID {
+    const binary = Binary.fromExtendedJSON(doc, options);
+    if (binary.sub_type !== Binary.SUBTYPE_UUID) {
+      throw new TypeError(
+        `Binary EJSON passed to UUID containing with unsupported sub_type: "${binary.sub_type}".`
+      );
+    }
+    return new UUID(binary.buffer);
+  }
+
+  /** @internal */
+  static createPk(): UUID {
+    return new UUID();
+  }
+
+  /**
+   * Converts to a string representation of this Id.
+   *
+   * @returns return the 32(36) character hex string representation.
+   * @internal
+   */
+  [Symbol.for('nodejs.util.inspect.custom')](): string {
+    return this.inspect();
+  }
+
+  inspect(): string {
+    return `new UUID("${this.toHexString()}")`;
```

Comment:
Is it "normal" / convention to prefix this sting with `new`?

Reply:
I believe so, this is how it looks in `ObjectId` - and was added/refactored late 2020.